KeyVaultEntityRecoverySamples.cs

```diff
@@ -79,64 +83,41 @@ public static async Task DemonstrateRecoveryAndPurgeAsync()
+                RecoverDeletedSecretOperation recoverDeletedSecretOperation = await secretClient.StartRecoverDeletedSecretAsync(secretName);
+                await recoverDeletedSecretOperation.WaitForCompletionAsync();
                 Console.WriteLine("done.");
 
                 // confirm recovery
                 Console.Write("Retrieving recovered secret...");
-                await RetryHttpRequestAsync(
-                    async () => { return retrievedSecretResponse = await sample.DataClient.GetSecretWithHttpMessagesAsync(vaultUri, secretName, secretVersion: String.Empty).ConfigureAwait(false); },
-                    "recover deleted secret",
-                    SampleConstants.RetryPolicies.DefaultSoftDeleteRetryPolicy)
-                    .ConfigureAwait(false);
+                await secretClient.GetSecretAsync(secretName);
                 Console.WriteLine("done.");
 
                 // delete secret
-                Console.Write("Deleting secret (pass #2)...");
-                await sample.DataClient.DeleteSecretWithHttpMessagesAsync(vaultUri, secretName).ConfigureAwait(false);
+                Console.Write("Deleting recorvered secret...");
+                DeleteSecretOperation deleteRecoveredSecretOperation = await secretClient.StartDeleteSecretAsync(secretName);
+                await deleteRecoveredSecretOperation.WaitForCompletionAsync();
                 Console.WriteLine("done.");
 
                 // retrieve deleted secret
-                Console.Write("Retrieving the deleted secret (pass #2)...");
-                await RetryHttpRequestAsync(
-                    async () => { return deletedSecretResponse = await sample.DataClient.GetDeletedSecretWithHttpMessagesAsync(vaultUri, secretName); },
-                    "get deleted secret",
-                    SampleConstants.RetryPolicies.DefaultSoftDeleteRetryPolicy)
-                    .ConfigureAwait(false);
-                Console.WriteLine("done.");
-
-                // purge secret
-                Console.Write("Purging deleted secret...");
-                await sample.DataClient.PurgeDeletedSecretWithHttpMessagesAsync(vaultUri, secretName).ConfigureAwait(false);
+                Console.Write("Retrieving the deleted secret...");
+                await secretClient.GetDeletedSecretAsync(secretName);
                 Console.WriteLine("done.");
             }
-            catch (KeyVaultErrorException kvee)
-            {
-                Console.WriteLine("Unexpected KeyVault exception encountered: {0}", kvee.Message);
+            catch (RequestFailedException ex)
 
-                throw;
-            }
-            catch (CloudException ce)
             {
-                Console.WriteLine("Unexpected ARM exception encountered: {0}", ce.Message);
+                Console.WriteLine("Unexpected KeyVault exception encountered: {0}", ex.Message);
```

Comment:
Nit: the product name is "Key Vault" (with a space). We should use that in customer-facing messages / docs.

Reply:
Updated.